exporter/prometheusexporter/prometheus_test.go

```diff
@@ -681,7 +681,6 @@ this_one_there_where_{arch="x86",instance="test-instance",job="test-service",os=
 			// Scrape metrics, with the Accept header set to the value specified in the test case
 			req, err := http.NewRequest(http.MethodGet, "http://"+addr+"/metrics", http.NoBody)
 			require.NoError(t, err)
-			req.Close = true
```

Comment:
This was an early attempt to fix this flakiness, but it wasn't enough, removing it. For reference it was introduced via #42080